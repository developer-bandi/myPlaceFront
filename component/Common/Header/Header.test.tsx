import { act, renderHook } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import useHeader from "./HeaderHook";
import React, { ReactNode } from "react";

describe("Header Hook 테스트", () => {
  const mockStore = configureMockStore()({
    userLogin: {
      loading: false,
      error: false,
    },
    mypageModal: {
      active: false,
    },
  });
  const wrapper = ({ children }: { children: ReactNode }) => (
    <Provider store={mockStore}>{children}</Provider>
  );

  it("로그인 체크하는 useEffect 테스트", () => {
    renderHook(() => useHeader(), { wrapper });
    expect(mockStore.getActions()[0]).toEqual({
      payload: undefined,
      type: "userLogin/checkSignin",
    });
  });

  it("changePageModal 함수 테스트", () => {
    const { result } = renderHook(() => useHeader(), { wrapper });
    act(() => {
      result.current.changePageModal();
    });
    expect(mockStore.getActions()[2]).toEqual({
      payload: undefined,
      type: "modalStatus/setMypage",
    });
    expect(mockStore.getActions()[3]).toEqual({
      payload: false,
      type: "modalStatus/setNotice",
    });
  });
  it("changeNoticeModal 함수 테스트", () => {
    const { result } = renderHook(() => useHeader(), { wrapper });
    act(() => {
      result.current.changeNoticeModal();
    });

    expect(mockStore.getActions()[5]).toEqual({
      payload: undefined,
      type: "modalStatus/setNotice",
    });
    expect(mockStore.getActions()[6]).toEqual({
      payload: false,
      type: "modalStatus/setMypage",
    });
  });
});
