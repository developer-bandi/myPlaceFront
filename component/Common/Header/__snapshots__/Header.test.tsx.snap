// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Presentational 테스트 반응형 테스트 모바일인 경우 1`] = `
<div>
  <header
    class="mainBlock"
  >
    <div
      class="subBlock"
    >
      <div
        class="title"
      >
        <a
          href="/"
        >
          MyPlace
        </a>
      </div>
      <ul
        class="navigationBlock"
      />
      <button
        class="signin"
        data-testid="logout"
      >
        <a
          href="/user/auth/signin"
        >
          로그인/회원가입
        </a>
      </button>
    </div>
    <div
      class="subBlock"
      data-testid="mobile"
    >
      <ul
        class="bottomNavigationBlock"
      >
        <li
          class="navigationName"
        >
          <a
            href="/findplace"
          >
            장소찾기
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/community/postlist"
          >
            커뮤니티
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/contribute"
          >
            기여하기
          </a>
        </li>
      </ul>
    </div>
  </header>
</div>
`;

exports[`Header Presentational 테스트 상황 테스트 로그인 안했을경우 1`] = `
<div>
  <header
    class="mainBlock"
  >
    <div
      class="subBlock"
    >
      <div
        class="title"
      >
        <a
          href="/"
        >
          MyPlace
        </a>
      </div>
      <ul
        class="navigationBlock"
      />
      <button
        class="signin"
        data-testid="logout"
      >
        <a
          href="/user/auth/signin"
        >
          로그인/회원가입
        </a>
      </button>
    </div>
    <div
      class="subBlock"
      data-testid="mobile"
    >
      <ul
        class="bottomNavigationBlock"
      >
        <li
          class="navigationName"
        >
          <a
            href="/findplace"
          >
            장소찾기
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/community/postlist"
          >
            커뮤니티
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/contribute"
          >
            기여하기
          </a>
        </li>
      </ul>
    </div>
  </header>
</div>
`;

exports[`Header Presentational 테스트 상황 테스트 로그인 하고 마이페이지 창을 열였을 경우 1`] = `
<div>
  <header
    class="mainBlock"
  >
    <div
      class="subBlock"
    >
      <div
        class="title"
      >
        <a
          href="/"
        >
          MyPlace
        </a>
      </div>
      <ul
        class="navigationBlock"
      />
      <button
        class="signin"
        data-testid="logout"
      >
        <a
          href="/user/auth/signin"
        >
          로그인/회원가입
        </a>
      </button>
    </div>
    <div
      class="subBlock"
      data-testid="mobile"
    >
      <ul
        class="bottomNavigationBlock"
      >
        <li
          class="navigationName"
        >
          <a
            href="/findplace"
          >
            장소찾기
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/community/postlist"
          >
            커뮤니티
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/contribute"
          >
            기여하기
          </a>
        </li>
      </ul>
    </div>
    <div
      class="myPageNavigationBlock"
      data-testid="mypageModal"
    >
      <div
        class="mainBlock"
      >
        <nav
          class="subBlock"
        >
          <ul
            class="navListBlock"
          >
            <li
              class="nav"
              data-testid="modalActvieChange0"
            >
              <a
                href="/user/mypage/bookmark"
              >
                북마크
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange1"
            >
              <a
                href="/user/mypage/review"
              >
                작성 후기
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange2"
            >
              <a
                href="/user/mypage/post"
              >
                작성 글
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange3"
            >
              <a
                href="/user/mypage/comment"
              >
                작성 댓글
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange4"
            >
              <a
                href="/user/mypage/infosetting"
              >
                개인정보관리
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange5"
            >
              <a
                href="/user/mypage/password"
              >
                비밀번호 변경
              </a>
            </li>
            <li
              class="nav"
              data-testid="modalActvieChange6"
            >
              <a
                href="/user/mypage/withdrawal"
              >
                회원탈퇴
              </a>
            </li>
            <li
              class="logout"
              data-testid="userLogout"
            >
              로그아웃
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header Presentational 테스트 상황 테스트 로그인 했을 경우 1`] = `
<div>
  <header
    class="mainBlock"
  >
    <div
      class="subBlock"
    >
      <div
        class="title"
      >
        <a
          href="/"
        >
          MyPlace
        </a>
      </div>
      <ul
        class="navigationBlock"
      />
      <div
        class="userBlock"
        data-testid="login"
      >
        <p
          class="userInfo"
        >
          test
          님 반갑습니다
        </p>
        <button
          class="mypage"
          data-testid="modalActvieChange"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2A10.13 10.13 0 0 0 2 12a10 10 0 0 0 4 7.92V20h.1a9.7 9.7 0 0 0 11.8 0h.1v-.08A10 10 0 0 0 22 12 10.13 10.13 0 0 0 12 2zM8.07 18.93A3 3 0 0 1 11 16.57h2a3 3 0 0 1 2.93 2.36 7.75 7.75 0 0 1-7.86 0zm9.54-1.29A5 5 0 0 0 13 14.57h-2a5 5 0 0 0-4.61 3.07A8 8 0 0 1 4 12a8.1 8.1 0 0 1 8-8 8.1 8.1 0 0 1 8 8 8 8 0 0 1-2.39 5.64z"
            />
            <path
              d="M12 6a3.91 3.91 0 0 0-4 4 3.91 3.91 0 0 0 4 4 3.91 3.91 0 0 0 4-4 3.91 3.91 0 0 0-4-4zm0 6a1.91 1.91 0 0 1-2-2 1.91 1.91 0 0 1 2-2 1.91 1.91 0 0 1 2 2 1.91 1.91 0 0 1-2 2z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="subBlock"
      data-testid="mobile"
    >
      <ul
        class="bottomNavigationBlock"
      >
        <li
          class="navigationName"
        >
          <a
            href="/findplace"
          >
            장소찾기
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/community/postlist"
          >
            커뮤니티
          </a>
        </li>
        <li
          class="navigationName"
        >
          <a
            href="/contribute"
          >
            기여하기
          </a>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
